### PR TITLE
SelfHostedEtcd: updates to use beta of etcd operator

### PR DESCRIPTION
**Pull request by hongchaodeng**
This PR updates etcd operator spec to beta and updates SHE code path for it.

### CODE REVIEW

**Comment by xiang90** [comment]
rktbot run etcd tests

**Comment by xiang90** [comment]
rktbot run tests

**Comment by hongchaodeng** [comment]
Failed SHE test also failed on non-SHE test:
```
2017-02-22T22:43:46Z kola: --- FAIL: experimentaletcd.smoke on gce (507.305s)
2017-02-22T22:43:46Z kola:         SSH session returned error for cmd sudo /usr/bin/rkt run \
                --stage1-name=coreos.com/rkt/stage1-fly:1.19.0 \
        	--volume home,kind=host,source=/home/core \
        	--mount volume=home,target=/core \
        	--volume manifests,kind=host,source=/etc/kubernetes/manifests \
        	--mount volume=manifests,target=/etc/kubernetes/manifests \
                --trust-keys-from-https \
		quay.io/coreos/bootkube-dev:c2f7a0de280760109dc6932f0ffc79b598875e12 --exec \
		/bootkube -- start --asset-dir=/core/assets --etcd-server=http://10.240.0.32:12379 --experimental-self-hosted-etcd: wait: remote command exited without exit status or exit signal
```
https://jenkins-tectonic.prod.coreos.systems/job/bootkube-dev/335/console
https://jenkins-tectonic.prod.coreos.systems/job/bootkube-SHE-dev/5/console

**Comment by hongchaodeng** [comment]
@aaronlevy @pbx0 Test passed now. Can you take a look at the PR?

**Comment by pbx0** [comment]
The tests LGTM to me. There has been a little flakyness I have seen when multiple PRs are running multiple test runs but re-running seems to have fixed it.

I'll also run this through the scale test I wrote and see if that test improves (this test isn't on the CI yet).

File: pkg/asset/internal/templates.go [hunk 1]
```diff
@@ -539,12 +540,16 @@ spec:
     spec:
       containers:
       - name: etcd-operator
-        image: quay.io/coreos/etcd-operator:c391d8b7638deb81aa877773a0acce389f602415
+        image: quay.io/coreos/etcd-operator:dev
```
Comment (pbx0) [inline pkg/asset/internal/templates.go]: Will this tag change at all? It'd be best to use a more specific release even if temporary.

Comment (xiang90) [inline pkg/asset/internal/templates.go]: we will change this to release 0.2.1 tomorrow. we just want to make sure current etcd operator works with bootkube as it is now. then we bump it, then we change it here. it is kind of chicken egg problem for now.

Comment (hongchaodeng) [inline pkg/asset/internal/templates.go]: This has some offline discussion as context.

First of all, latest beta release (the only beta release) has some client-go conflict.

We don't have nightly job to push to this tag yet, so we manually test latest etcd operator, build and push this tag.

We will change this to v0.2.1 once it's released, and keep this tag unchanged before that.

**Comment by hongchaodeng** [comment]
Updated etcd operator image to v0.2.1

**Comment by xiang90** [comment]
rktbot run etcd tests

File: Makefile [hunk 2]
```diff
@@ -67,9 +67,9 @@ vendor:
 	@mv $@/k8s.io/kubernetes/vendor/* $(abspath $@)
 	@cd $@/k8s.io/ && ln -sf kubernetes/staging/src/k8s.io/client-go client-go
 	@rm -rf $@/k8s.io/kubernetes/vendor $@/k8s.io/kubernetes/.git
-	@echo "vendoring etcd-operator spec"
+	@echo "vendoring etcd-operator spec: ${ETCD_OPERATOR_VERSION}"
 	@git clone https://github.com/coreos/etcd-operator /tmp/etcd-operator > /dev/null 2>&1
-	@cd /tmp/etcd-operator && git checkout $(ETCD_OPERATOR_VERSION) > /dev/null 2>&1
+	@cd /tmp/etcd-operator && git checkout ${ETCD_OPERATOR_VERSION} > /dev/null 2>&1
```
Comment (aaronlevy) [inline Makefile]: `$()` was correct for referencing a makefile variable

File: pkg/asset/internal/templates.go [hunk 3]
```diff
@@ -523,6 +523,7 @@ spec:
     protocol: TCP
 `)
 
+	// TODO: switch to official release quay image after v0.2.1 release
```
Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: Looks like this TODO isn't necessary anymore?

File: Makefile [hunk 4]
```diff
@@ -67,9 +67,9 @@ vendor:
 	@mv $@/k8s.io/kubernetes/vendor/* $(abspath $@)
 	@cd $@/k8s.io/ && ln -sf kubernetes/staging/src/k8s.io/client-go client-go
 	@rm -rf $@/k8s.io/kubernetes/vendor $@/k8s.io/kubernetes/.git
-	@echo "vendoring etcd-operator spec"
+	@echo "vendoring etcd-operator spec: ${ETCD_OPERATOR_VERSION}"
```
Comment (aaronlevy) [inline Makefile]: This also can use the `$()` format for consistency

File: Makefile (hunk 2 above)
Comment (hongchaodeng) [inline Makefile]: Weird.. This also works.. I'm fine o change.

**Comment by xiang90** [comment]
@pbx0 @aaronlevy 

test is failing: https://jenkins-tectonic.prod.coreos.systems/job/bootkube-dev/348/consoleText

i do not think it is relevant to this pr though.

**Comment by pbx0** [comment]
rktbot run etcd tests

**Comment by pbx0** [comment]
rktbot run tests

**Comment by xiang90** [comment]
rktbot run etcd tests

**Comment by xiang90** [comment]
rktbot run etcd tests

**Comment by hongchaodeng** [comment]
rktbot run etcd tests

**Comment by hongchaodeng** [comment]
rktbot run tests

**Comment by hongchaodeng** [comment]
failed:
```
2017-02-23T22:56:04Z kola: --- FAIL: bootkube.smoke on gce (734.630s)
2017-02-23T22:56:04Z kola:         ssh journalctl failed: dial tcp 104.198.50.54:22: getsockopt: connection refused
```

```
2017-02-23T22:58:41Z kola: --- FAIL: experimentaletcd.scale on gce (898.459s)
2017-02-23T22:58:41Z kola:         expected 1 etcd pods got 3: [Running Running Running]
```

**Comment by hongchaodeng** [comment]
rktbot run tests

**Comment by hongchaodeng** [comment]
Known failure:
```
2017-02-24T01:13:11Z kola: --- FAIL: bootkube.destruct.delete on gce (404.560s)
2017-02-24T01:13:11Z kola:         nginxCheck: kubectl:error: Scaling the resource failed with: Get https://104.154.136.150:443/apis/extensions/v1beta1/namespaces/default/replicasets/my-nginx-379829228: dial tcp 104.154.136.150:443: getsockopt: connection refused; Current resource version Unknown
```

**Comment by hongchaodeng** [comment]
rktbot run tests

**Comment by pbx0** [comment]
rktbot run etcd tests

**Comment by pbx0** [comment]
rktbot run etcd tests

**Comment by hongchaodeng** [comment]
rktbot run tests

**Comment by hongchaodeng** [comment]
rktbot run tests

**Comment by xiang90** [comment]
rktbot run tests

**Comment by pbx0** [comment]
rktbot run tests

**Comment by xiang90** [comment]
rktbot run tests

**Comment by xiang90** [comment]
@pbx0 @aaronlevy Can you please take another look? Thanks!

**Comment by aaronlevy** [comment]
Yeah, I'm testing this now

**Comment by aaronlevy** [comment]
One thing that I'm seeing is that `boot-etcd` never exits (and the static manifest exists after bootstrap)

**Comment by aaronlevy** [comment]
And after manually removing the boot-etcd manifest, the cluster seems to die. Might have a pivot issue still

**Comment by hongchaodeng** [comment]
@aaronlevy 

What went wrong? I tried out this branch again on multinode and self hosted cluster was created successfully. Let me rebase and try again.

**Comment by hongchaodeng** [comment]
I tried rebase and also worked.

What's the issue? Can we help resolve it?

**Pull request merged**